### PR TITLE
vim-patch:8.1.1458: crash when using gtags

### PR DESCRIPTION
**Pull request by janlazo**
Does this to apply Neovim's grid functions?

### CODE REVIEW

**Comment by bfredl** [comment]
Ideally this should be `abort()` and not `return;`. But maybe it is too risky without checking every usage in the code base. Perhaps we can do this for now and change to `abort();` right after the release.

File: src/nvim/screen.c [hunk 1]
```diff
@@ -5372,8 +5372,11 @@ void grid_puts_len(ScreenGrid *grid, char_u *text, int textlen, int row,
 
   screen_adjust_grid(&grid, &row, &col);
 
-  // safety check
-  if (grid->chars == NULL || row >= grid->Rows || col >= grid->Columns) {
+  // Safety check. The check for negative row and column is to fix issue
+  // #4102. TODO: find out why row/col could be negative.
```
Comment (bfredl) [inline src/nvim/screen.c]: should be `vim/vim#4102`. But IIRC we normally don't mention github issues in C code, the mention in the commit message is enough.

**Comment by blueyed** [comment]
> Ideally this should be `abort()` and not `return;`. But maybe it is too risky without checking every usage in the code base. Perhaps we can do this for now and change to `abort();` right after the release.

We could have an `assert` then now already?

**Comment by bfredl** [comment]
Then it would still abort in a RelWithDebInfo build, which we recommend to users.

**Pull request merged**